src/frontend/src/optimizer/plan_node/logical_hop_window.rs

```diff
@@ -72,6 +73,22 @@ impl LogicalHopWindow {
     ) -> PlanRef {
         Self::new(input, time_col, window_slide, window_size).into()
     }
+
+    pub fn window_start_col_idx(&self) -> usize {
+        self.schema().len() - 2
+    }
+
+    pub fn window_end_col_idx(&self) -> usize {
+        self.schema().len() - 1
+    }
+
+    pub fn o2i_col_mapping(&self) -> ColIndexMapping {
+        ColIndexMapping::identity_or_none(self.schema().len(), self.input.schema().len())
+    }
+
+    pub fn i2o_col_mapping(&self) -> ColIndexMapping {
+        ColIndexMapping::identity_or_none(self.input.schema().len(), self.schema().len())
```

Comment:
`identity` is enough here?

Reply:
yes, and the added `window_start` and `window_end` have not mapping to the input columns.

Reply:
So why identity_or_none instead of identity?

Reply:
in fact, I think the `identity_or_none` actually means `identity_with_target_size`, which means it will be identity for intersection of [0..source_size] and [0..target_size], and there will be no mapping for others